shark_turbine/kernel/wave/utils.py

```diff
@@ -0,0 +1,35 @@
+from ..compiler.ir import (
+    builtin_d,
+    InsertionPoint,
+    Location,
+    Operation,
+    transform_d,
+    UnitAttr,
+)
+
+from iree.compiler.dialects.transform import (
+    interpreter as transform_interpreter,
+    any_op_t,
+)
+
+
+def canonicalize_module(module: Operation):
+    with module.context, Location.unknown():
+        transform_module = builtin_d.Module.create()
+        transform_module_op = module.operation
+        transform_module_op.attributes["transform.with_named_sequence"] = UnitAttr.get()
+        with InsertionPoint(transform_module.body):
+            named_sequence = transform_d.NamedSequenceOp(
+                "__transform_main", [any_op_t()], []
+            )
+            with InsertionPoint(named_sequence.body):
```

Comment:
Just curious, is this the best way to setup canonicalization? by setting up a TD-like structure to do the canonicalization? I'd assume we'd have better API support from upstream to call canonicalization and or other standard patterns on.

Reply:
Good question and I was thinking about this myself. I think we can avoid TD by using other patterns, but there could be some advantages to using TD for now (as it has support for other patterns like LICM etc.). If we discover that we don't need that, we can always rewrite this without using TD.